### PR TITLE
Commands QA review

### PR DESCRIPTION
**Pull request by virtuald**
This is going to be a big one, going through and nit-picking/updating stuff. Will unmark draft once finished.

### CODE REVIEW

File: commands2/sequentialcommandgroup.py [hunk 1]
```diff
@@ -68,16 +76,13 @@ def execute(self):
...
-        if not self._commands:
-            return
-        if not self._currentCommandIndex > -1:
-            return
-        if not self._currentCommandIndex < len(self._commands):
-            return
+        if (
+            interrupted
+            and self._commands
+            and len(self._commands) > self._currentCommandIndex > -1
```
Comment (TheTripleV) [inline commands2/sequentialcommandgroup.py:82]:
```suggestion
            and -1 < self._currentCommandIndex < len(self._commands)
```

**Comment by virtuald** [comment]
Still need to do trigger and the scheduler, but overall everything looks pretty good so far. Bedtime.

File: commands2/commandscheduler.py [hunk 2]
```diff
@@ -29,7 +38,7 @@ def __new__(cls) -> CommandScheduler:
         return cls._instance
 
     @staticmethod
-    def getInstance() -> "CommandScheduler":
+    def getInstance() -> CommandScheduler:
```
Comment (TheTripleV) [inline commands2/commandscheduler.py:41]: Does this work in the oldest python we support?

Comment (virtuald) [reply commands2/commandscheduler.py:41]: The tests are running in 3.8 in CI so... yes? Isn't that what `from __future__ import annotations` does?

**Pull request merged**